client/session.go

```diff
@@ -170,8 +192,38 @@ func (s *session) setTopologyMap(topologyMap m3db.TopologyMap) error {
 	s.Lock()
 	prev := s.queues
 	s.queues = queues
-	s.setWriteOpPoolsWithLock(topologyMap)
 	s.topoMap = topologyMap
+	prevReplicas := s.replicas
+	s.replicas = topologyMap.Replicas()
+	s.quorum = topologyMap.QuorumReplicas()
+	if s.fetchBatchOpArrayArrayPool == nil ||
+		s.fetchBatchOpArrayArrayPool.Entries() != len(queues) {
+		s.fetchBatchOpArrayArrayPool = newFetchBatchOpArrayArrayPool(
+			s.opts.GetFetchBatchOpPoolSize(),
```

Comment:
Yeah we can add a expected fetch op size somewhere... We really need to make these pools dynamic now or we'll have big issues in the future, it's too hard to keep so many numbers fine tuned, especially as over time your metric intake/workload will change and you don't necessarily want to redeploy your software each time.